src/yunohost/app.py

```diff
@@ -1366,34 +1451,66 @@ def _get_setting(settings, name):
         return s.split(',') if s else []
 
     for app in apps_list:
-        with open(APPS_SETTING_PATH + app['id'] + '/settings.yml') as f:
-            app_settings = yaml.load(f)
 
-            if 'no_sso' in app_settings:
-                continue
+        app_settings = read_yaml(APPS_SETTING_PATH + app['id'] + '/settings.yml')
+
+        if 'domain' not in app_settings:
+            continue
+        if 'path' not in app_settings:
+            continue
+
+        # This 'no_sso' settings sound redundant to not having $path defined ....
+        # At least from what I can see, all apps using it don't have a path defined ...
+        if 'no_sso' in app_settings:
+            continue
+
+        domain = app_settings['domain']
+        path = app_settings['path'].rstrip('/')
+
+        def _sanitized_absolute_url(perm_url):
+            # Nominal case : url is relative to the app's path
+            if perm_url.startswith("/"):
+                perm_domain = domain
+                perm_path = path + perm_url.rstrip("/")
+            # Otherwise, the urls starts with a domain name, like domain.tld/foo/bar
+            # We want perm_domain = domain.tld and perm_path = "/foo/bar"
+            else:
+                perm_domain, perm_path = perm_url.split("/", 1)
+                perm_path = "/" + perm_path.rstrip("/")
+
+            return perm_domain + perm_path
+
+        # Skipped
+        skipped_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'skipped_uris')]
+        skipped_regex += _get_setting(app_settings, 'skipped_regex')
+
+        # Redirected
+        redirected_urls.update(app_settings.get('redirected_urls', {}))
+        redirected_regex.update(app_settings.get('redirected_regex', {}))
+
+        # Legacy permission system using (un)protected_uris and _regex managed in app settings...
+        unprotected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'unprotected_uris')]
+        protected_urls += [_sanitized_absolute_url(uri) for uri in _get_setting(app_settings, 'protected_uris')]
+        unprotected_regex += _get_setting(app_settings, 'unprotected_regex')
+        protected_regex += _get_setting(app_settings, 'protected_regex')
```

Comment:
Are you sure that you can't have multiple `protected_regex` and `unprotected_regex` ?

Reply:
These are already lists, and `+=` "extends" the list, it doesn't just append one value. 

e.g.

```
L = [1,2]
L += [3,4]
```

results in `L == [1,2,3,4]`

Reply:
Oh, yes. I didn't see that `_get_setting` a list...